### PR TITLE
Use current extension for specifying IG resource format

### PR DESCRIPTION
**Pull request by jafeltra**
According to Zulip [here](https://chat.fhir.org/#narrow/stream/215610-shorthand/topic/How.20do.20I.20get.20SUSHI.20to.20ignore.20a.20binary.20JSON.20logical.20instance.3F/near/418612384), the extension URL we had been using to configure logical model examples has been deprecated. The new extension for the implementationguide-resource-format extension is `http://hl7.org/fhir/tools/StructureDefinition/implementationguide-resource-format`.

This PR updates SUSHI to automatically add the correct extension when creating the `resource` entry of an `Instance` of a `Logical` for the IG JSON. It also adds a warning when a provided configuration uses the deprecated extension URL to help users update to the current URL.

I ran a full regression and no one seems to actually be using this old URL because I didn't see any new warnings logged.

Note that there is already a PR on FSH School to update the extension we document there.

### CODE REVIEW

**Review by cmoesel — COMMENTED** [review]
In reviewing this, I discovered a bug in our general implementation of this for bring-your-own-JSON logical instances. In short, it only works if the `resourceType` in the JSON is the `id` of the logical model -- but if the `resourceType` is the canonical URL of the logical model, it doesn't work and ends up creating two entries for the instance in the IG JSON (and also doesn't flag the deprecated URL). I think that using the canonical URL is actually the _right_ thing, so it's unfortunate we don't handle it correctly. (Note that when we export FSH Instances of logical models, we use the canonical URL).

So... I think we need to fix that. For legacy purposes, we should probably continue to support the id-based resourceType, but we also need to recognize and accept the canonical URL based resourceType. The question is if we put it in this PR or if we track it separately. I think if the solution is fairly straight-forward, we should just put it in here. But if it's more than a few lines to change (in the production code), we should handle it separately. What do you think?

I'm attaching a project that demonstrates all the different flavors of this feature (including the broken ones): [LogicalInstanceExtensions.zip](https://github.com/FHIR/sushi/files/14117910/LogicalInstanceExtensions.zip)

File: src/ig/IGExporter.ts [hunk 1]
```diff
@@ -896,18 +906,24 @@ export class IGExporter {
...
     }
+    if (
+      pkgResource instanceof InstanceDefinition &&
+      pkgResource._instanceMeta.sdKind === 'logical' &&
+      configResource?.extension?.some(ext => ext.url === DEPRECATED_RESOURCE_FORMAT_EXTENSION)
+    ) {
+      logger.warn(
+        `The extension ${DEPRECATED_RESOURCE_FORMAT_EXTENSION} has been deprecated. Update the configuration for ${configResource.reference?.reference ?? newResource.name} to use the current extension, ${CURRENT_RESOURCE_FORMAT_EXTENSION}.`
+      );
+    }
```
[the remark this answers is not in the document]
Comment (jafeltra) [reply src/ig/IGExporter.ts]: I agree! Updated in [6f0d26b](https://github.com/FHIR/sushi/pull/1421/commits/6f0d26b9de9e9d0747e07f6cc26987f62804375e).

**Comment by jafeltra** [comment]
> In reviewing this, I discovered a bug in our general implementation of this for bring-your-own-JSON logical instances. In short, it only works if the resourceType in the JSON is the id of the logical model -- but if the resourceType is the canonical URL of the logical model, it doesn't work and ends up creating two entries for the instance in the IG JSON (and also doesn't flag the deprecated URL). I think that using the canonical URL is actually the right thing, so it's unfortunate we don't handle it correctly. (Note that when we export FSH Instances of logical models, we use the canonical URL).

I had noticed this recently too, so since you ran into it as well, I think it is worth fixing sooner rather than later. I think I handled it correctly, and it wasn't a huge change, so I included it in [c0e6ead](https://github.com/FHIR/sushi/pull/1421/commits/c0e6ead15624ed80b8891ff29a8a7d247f211180). I think it fixes all the cases in the example you provided, but let me know if I missed anything.

**Review by cmoesel — APPROVED** [review]
Yep. That did the trick! Nice job.

**Review by mint-thompson — CHANGES_REQUESTED** [review]
This looks all correct! I have one very minor change to suggest.

File: src/ig/IGExporter.ts [hunk 2]
```diff
@@ -897,18 +907,24 @@ export class IGExporter {
...
-        url: 'http://hl7.org/fhir/StructureDefinition/implementationguide-resource-format',
-        valueCode: 'application/fhir+json'
-      });
+      if (
+        !configResource?.extension?.some(ext => IG_RESOURCE_FORMAT_EXTENSIONS.includes(ext.url))
+      ) {
+        // Logical instances should add a special extension. See: https://fshschool.org/docs/sushi/tips/#instances-of-logical-models
+        newResource.extension = newResource.extension ?? [];
+        newResource.extension.push({
+          url: 'http://hl7.org/fhir/tools/StructureDefinition/implementationguide-resource-format',
```
[the remark this answers is not in the document]
Comment (jafeltra) [reply src/ig/IGExporter.ts]: Sure seems like a good idea! Added in [e2c0c25](https://github.com/FHIR/sushi/pull/1421/commits/e2c0c258a83d05865963f6ab70884e8ab9a91f0a).

**Review by mint-thompson — APPROVED** [review]

**Review by cmoesel — APPROVED** [review]
Still looks good to me! (In fact, even better!)

**Pull request merged**